adlfs/multithread.py

```diff
@@ -327,7 +327,7 @@ def run(self, nthreads=None, monitor=True):
         for (rfile, lfile), dic in self.progress.items():
             unique = dic['uuid']
             if len(dic['waiting']) > 1:
-                parts = [self.temp_upload_path+unique+"_%i" % i for i
+                parts = [self.temp_upload_path+unique+"/part_%i" % i for i
```

Comment:
It created one folder (identifier `unique`, above) for each multi-chunk file, into which all the parts go. Interestingly, it seems there is no need to explicitly create the folder first.